File: demo/src/main/java/com/example/demo/domain/entity/Food.java
package com.example.demo.domain.entity;

import javax.persistence.*;
import java.math.BigDecimal;

@Entity
public class Food extends BaseEntity {


    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id")
    private FoodCategory foodCategory;
    private String name;

    private BigDecimal price;

    public Food() {
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public BigDecimal getPrice() {
        return price;
    }

    public void setPrice(BigDecimal price) {
        this.price = price;
    }

    public FoodCategory getFoodCategory() {
        return foodCategory;
    }

    public void setFoodCategory(FoodCategory foodCategory) {
        this.foodCategory = foodCategory;
    }
}

File: food-guro-client/src/app/food.ts
export class Food {
    id: number;
    foodCategory: string;
    name: string;
    price: number;
    description: string;

    public constructor(id: number, foodCategory: string, name: string, price: number, description: string) {
        this.id = id;
        this.foodCategory = foodCategory;
        this.name = name;
        this.price = price;
        this.description = description;
    }


}